### PR TITLE
Fixed the training report library, Fixed the issue around duplicate staff and mandatory training fixes

### PR DESCRIPTION
**Pull request by aaron-russell**

### CODE REVIEW

**Comment by aaron-russell** [comment]
[Training Report encoding issue](https://trello.com/c/GyTQ4IYk/2-training-report-encoding-issue)

**Comment by aaron-russell** [comment]
[Mandatory Training on Training Report](https://trello.com/c/22rOAKVx/3-mandatory-training-on-training-report)

File: server/routes/reports/trainingReport/training.js [hunk 1]
```diff
@@ -127,24 +134,13 @@ const updateProps = 'JobName,Category,Title,Expires'.split(',');
...
-          }
-        });
-        if (!foundTrn) {
-          reportData.push({ ID: value.ID, NameOrIdValue: value.NameOrIdValue, MandatoryCount: 0, NonMandatoryCount: 0, Count: 0});
-        }
-      }
-    });
-  }
+const createExpireExpiringData = async (trainingData) => {
+  return trainingData.filter((trainingWorker, index, self) => self.findIndex(t => trainingWorker.ID === t.ID) === index).map(
```
Comment (chadtomkiss-madetech) [inline server/routes/reports/trainingReport/training.js:138]: Lovveee this!

Formatting isn't something that should be debated in a PR, but here we have `.filter` and `.map` inline which is hard to parse.

Are we using Prettier? There's a `.prettierrc.yml` but I'm not sure it's being used.

As an example, this is what it would do:

```
return trainingData
  .filter(
    (trainingWorker, index, self) =>
      self.findIndex((t) => trainingWorker.ID === t.ID) === index
  )
  .map((training) => {
    return {
      ID: training.ID,
      NameOrIdValue: training.NameOrIdValue,
      MandatoryCount: 0,
      NonMandatoryCount: 0,
      Count: 0,
    };
  });
```

It's a lot easier to see what functions are being applied and even what the object structure looks like.

File: server/routes/reports/trainingReport/training.js [hunk 2]
```diff
@@ -132,12 +132,20 @@ const updateProps = 'JobName,Category,Title,Expires'.split(',');
  * Helper Function used to create expired/expiring traing data
  *
  * @param {Object} trainingData
- * @param {Object} All customized expired/expiring training report data
+ * @return {Object} All workers with traiing
```
Comment (chadtomkiss-madetech) [inline server/routes/reports/trainingReport/training.js:135]: Fix typo 
```suggestion
 * @return {Object} All workers with training
```
